tempodb/encoding/v2/create_block.go

```diff
@@ -0,0 +1,107 @@
+package v2
+
+import (
+	"context"
+	"io"
+	"math"
+	"time"
+
+	"github.com/google/uuid"
+	model_v2 "github.com/grafana/tempo/pkg/model/v2"
+	"github.com/grafana/tempo/pkg/tempopb"
+	"github.com/grafana/tempo/tempodb/backend"
+	"github.com/grafana/tempo/tempodb/encoding/common"
+	"github.com/pkg/errors"
+)
+
+const DefaultFlushSizeBytes int = 30 * 1024 * 1024 // 30 MiB
+
+func CreateBlock(ctx context.Context, cfg *common.BlockConfig, tenantID string, blockID uuid.UUID, encoding backend.Encoding,
+	dataEncoding string, estimatedTotalObjects int, i common.TraceIterator, to backend.Writer) (*backend.BlockMeta, error) {
+	defer i.Close()
+
+	newMeta := backend.NewBlockMeta(tenantID, blockID, VersionString, encoding, model_v2.Encoding)
+
+	dec := model_v2.NewSegmentDecoder()
```

Comment:
should we use `model.CurrentEncoding`? i'm honestly not sure. we've definitely made it quite difficult to add a v3 encoding, but i think that's ok. once a completely new block format hits all this code will be retired.